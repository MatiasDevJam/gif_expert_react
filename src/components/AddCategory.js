import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    // esta función nos permite cambiar el estado del input
    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    // función que permite quitar el comportamiento por defecto del form
    const handleSubmit = ( e ) => {
        e.preventDefault();

        //validamos que el usuario no envie datos vacios
        if( inputValue.trim().length > 2 ){
            setCategories( cats => [inputValue, ...cats ]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input  
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

// a traves de props nos aseguramos que setCategories sea requerido
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}