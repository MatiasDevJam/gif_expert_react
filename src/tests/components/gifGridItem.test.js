import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';     

describe('Pruebas en <GifGridItem />', ()=> {

    const title = 'Un título';
    const url = 'https://localhost/unaimagen.jgp';

    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

    test('Dede de mostrar el componente correctamente', ()=>{

        expect( wrapper ).toMatchSnapshot();
    })

    test('Dede de mostrar en el párrafo un title', ()=>{

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('Dede tener la imagen igual a la url y alt de los props', ()=>{

        const img = wrapper.find('img');
        expect( img.prop('alt') ).toBe( title );
        expect( img.prop('src') ).toBe( url );
    })

    test('El div debe tener la clase animate__backInDown', ()=>{

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__backInDown') ).toBe( true );
        
    })

})