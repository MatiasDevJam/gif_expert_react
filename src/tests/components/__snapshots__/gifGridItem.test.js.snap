// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Dede de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInDown"
>
  <img
    alt="Un título"
    src="https://localhost/unaimagen.jgp"
  />
  <p>
    Un título
  </p>
</div>
`;
